### ISSUE TITLE
The debug gem should not use RubyVM::*

### ISSUE THREAD

**Issue by eregon** [ruby/debug#887]
This gem currently [uses `RubyVM::`](https://github.com/search?q=repo%3Aruby%2Fdebug%20RubyVM&type=code) which is of course not portable and just CRuby internals.
As a standard gem for debugging, it should steer away from that and only use public and well-supported debugging APIs, not CRuby internals (which RubyVM is).
Otherwise, we'd be back to not using a proper public debugging APIs like all the previous debugger gems before, which eventually died due to depending on internals.

**Comment by ko1** [comment]
Please propose enough APIs to (C)Ruby at first. But I don't think it is feasible.
Anyway, we can use implementation specific code if available.

**Comment by eregon** [comment]
Could you summarize the current usages of RubyVM:: and what are they needed for?

Also this shouldn't be closed as `completed` because it's clearly not fixed. I think it makes more sense to leave it open until it's fixed.

**Comment by ko1** [comment]
> because it's clearly not fixed.

debug.gem doesn't hesitate to use any implementation dependent feature if it maintainable (and I can do it on `::RubyVM`). So it is not an issue.

**Comment by ko1** [comment]
> Could you summarize the current usages of RubyVM:: and what are they needed for?

Maybe biggest point is ISeq structure to find out breakpoint positions by the line. https://github.com/ruby/debug/blob/master/lib/debug/breakpoint.rb#L256
